iproute/iproute.go

```diff
@@ -0,0 +1,105 @@
+// +build linux
+
+package iproute
+
+import (
+	"syscall"
+	"time"
+
+	"github.com/aporeto-inc/netlink-go/common"
+	"github.com/vishvananda/netlink"
+)
+
+// Iproute is the wrapper around netlinkHandle
+type Iproute struct {
+}
+
+// NewIPRouteHandle returns a reference IpRoute structure
+func NewIPRouteHandle() (IPRoute, error) {
+	return &Iproute{}, nil
+
+}
+
+// AddRule add rule to the rule table
+func (i *Iproute) AddRule(rule *netlink.Rule) error {
+	//mask of the high bits
+	seq := time.Now().Unix() & 0x00000000ffffffff
+	nlmsghdr := common.BuildNlMsgHeader(
+		syscall.RTM_NEWRULE,
+		syscall.NLM_F_ATOMIC|syscall.NLM_F_REQUEST|syscall.NLM_F_ACK|syscall.NLM_F_MATCH,
+		syscall.SizeofNlMsghdr)
+	nlmsghdr.Seq = uint32(seq)
+	rtmsgbuf := rtmsgToWire(syscall.AF_INET, uint8(rule.Table), syscall.RTPROT_BOOT, syscall.RTN_UNICAST)
+	priobuf := priorityAttrToWire(uint32(rule.Priority))
+	markbuf := markAttrToWire(uint32(rule.Mark))
+	maskbuf := markMaskAttrToWire(uint32(rule.Mask))
+	nlmsghdr.Len = syscall.SizeofNlMsghdr + uint32(len(rtmsgbuf)+len(priobuf)+len(markbuf)+len(maskbuf))
+	buf := common.SerializeNlMsgHdr(nlmsghdr)
+
```

Comment:
Make a variadic function allocate slice once and copy data. 
Multiple appends here are not great and the same code is present in all functions